proof_test.go

```diff
@@ -1016,6 +1016,47 @@ func TestProofOfAbsenceBorderCase(t *testing.T) {
 	}
 }
 
+func TestProofOfAbsenceBorderCaseReversed(t *testing.T) {
+	root := New()
+
+	key1, _ := hex.DecodeString("0001000000000000000000000000000000000000000000000000000000000001")
+	key2, _ := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000001")
```

Comment:
Note this new test case which is exactly the same of `TestProofOfAbsenceBorderCase` but with key1 and key2 reversed. (Note: at some point we can refactor tests and make it sub-tests; for now I'm interested in correctness).

This fails because `key2` is the absent one, and is processed first than `key1` thus we add an extension status. But considering how tree reconstruction works, it will skip `key1`. (It's complicated to explain here; so I can explain better in Matrix).